sui_core/src/authority_active.rs

```diff
@@ -30,19 +30,25 @@
 */
 
 use std::{collections::BTreeMap, sync::Arc};
-
 use sui_types::{base_types::AuthorityName, error::SuiResult};
 
 use crate::{
     authority::AuthorityState, authority_aggregator::AuthorityAggregator,
     authority_client::AuthorityAPI,
 };
 
+pub mod gossip;
+use gossip::gossip_process;
+
+#[cfg(test)]
+#[path = "unit_tests/gossip_tests.rs"]
```

Comment:
Ideally tests should be close to their source and not off in another directory. I'd like us to start moving away from this paradigm and instead just have tests in a test module nested (on the file system as well) under the module it's intended for.

Reply:
Ok, what directory hierarchy would you recommend?

Reply:
if there are only a few tests and the file is short then you can do:

```
#[cfg(test)]
mod test {
// Put tests here
}
```

If its long and you want tests in their own separate file then we can do something similar:
```
#[cfg(test)]
mod tests;
```
Where the directory hierarchy is: `src/gossip/tests.rs` and the module is at `src/gossip/mod.rs`